### PR TITLE
Rename all GraphQLBuilderExtensions classes to unique names

### PR DESCRIPTION
**Pull request by infogulch**
Take 2!  Sorry for the noise.  😄

### CODE REVIEW

**Comment by sungam3r** [comment]
In principle, I am not against such changes, although they will be useful in a very limited number of cases. Since these are breaking changes you should change target branch to `develop`. @Shane32 please review.

**Comment by infogulch** [comment]
I changed the target to the develop branch.

I agree, this does change the API surface so it is technically a breaking change.  Though, as you said, referencing these extension classes by name is `different from the "usual" use of the extension methods of these classes`, so it should not affect any user in practice.  In fact, the `using` statement I mentioned in #456 demonstrates that if any user *did* reference these classes by name they likely encountered a compiler error.

**Comment by sungam3r** [comment]
Now you have done a PR with 16 commits.

**Comment by infogulch** [comment]
I have redone the changes on top of the develop branch instead.

**Comment by infogulch** [comment]
Should we consider these changes a breaking change when no user could have referenced any of the changed APIs by name without encountering a compiler error?

**Comment by sungam3r** [comment]
It's binary breaking changes, yes.

File: src/Core/Extensions/GraphQLBuilderServerCoreExtensions.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ namespace GraphQL.Server
     /// <summary>
     /// GraphQL specific extension methods for <see cref="IGraphQLBuilder"/>.
     /// </summary>
-    public static class GraphQLBuilderExtensions
+    public static class GraphQLBuilderServerCoreExtensions
```
Comment (sungam3r) [inline src/Core/Extensions/GraphQLBuilderServerCoreExtensions.cs:17]:
```suggestion
    public static class GraphQLBuilderCoreExtensions
```

**Review by Shane32 — APPROVED** [review]

**Comment by infogulch** [comment]
Thank you for being patient while we worked through the issues on this PR!

**Review by sungam3r — APPROVED** [review]

**Pull request merged**